db/collection.go

```diff
@@ -105,7 +104,7 @@ func (c *Collection) FindByID(id core.InstanceID, opts ...TxnOption) (instance [
 
 // Create creates an instance in the collection.
 func (c *Collection) Create(v []byte, opts ...TxnOption) (id core.InstanceID, err error) {
-	_ = c.WriteTxn(func(txn *Txn) error {
+	err = c.WriteTxn(func(txn *Txn) error {
 		var ids []core.InstanceID
 		ids, err = txn.Create(v)
```

Comment:
This `ids, err =` is referencing the `err` that will be returned, so that `s/-/err` I think is redundant.
(Notice isn't `:=`).

I'm OK to leave this change if it makes it clearer in some sense.

Reply:
Not sure so sure about that actually. This 'error' was discovered because there _were_ no errors being propagated from this method, even when there should have been.

Reply:
If the error is in the actual Dispatch that arises out of the transaction, leaving out err won't catch that.

Reply:
What I mean is that `err` is the same reference as the return param of the Write func.
See [here](https://play.golang.org/p/Dnk6p38UmB5), which it's a simplified similar code.

Reply:
Ah, I see what you mean. You're correct, your change is necessary. 👍

Reply:
Github diffing hiding code is always tricky.